uvtools/dspec.py

```diff
@@ -1235,9 +1241,62 @@ def delay_filter_leastsq(data, flags, sigma, nmax, add_noise=False, freq_units =
     return mdl_array, cn_array, inp_data
 
 
-def sinc_downweight_mat_inv(nchan, df, filter_centers, filter_widths,
-                            filter_factors, cache = {}, wrap = False, wrap_interval=1,
-                            nwraps = 1000, no_regularization = False):
+def delay_interpolation_matrix(nchan, ndelay, wgts, fundamental_period=None, cache={}, taper='none', return_diagnostics=False):
+    '''
+    Computes a foreground interpolation matrix that, when applied to data,
+    interpolates over flagged channels with delays between -ndelays//2 and ndelays//2.
+
+    Parameters
+    ----------
+    nchan: integer
+        Number of frequency channels to interpolate over
+    ndelay,
+        number of delays to use in interpolation
+    wgts: float array
+        wgts to be applied to each frequency channel.
+    fundamental_period: float, optional
+        the fundamental period of reconstructed delays. Default: nchan
+    cache: dict, optional
+        optional cache holding pre-computed matrices
+    taper: string, optional
+        use a taper to fit.
+    Returns
+    ----------
+    (nchan, nchan) numpy array
+        that can be used to interpolate over channel gaps.
+    '''
+    assert len(wgts) == nchan, "nchan must equal length of wgts"
```

Comment:
Again, if we're making sure the user passed us something valid, we should not be doing it with an `assert` statement.

Reply:
i've changed these to if statements that raise value errors and added tests lines.